python/lsst/pipe/base/pipeline_graph/__main__.py

```diff
@@ -0,0 +1,583 @@
+# This file is part of pipe_base.
```

Comment:
We normally separate out the argument parser, the minimal main implementation, and the functionality into different places. This would allow the core functionality to at least be loaded in a test. There is no precedent in our code base for naming something like this `__main__.py`. This file is not even loaded by the tests to check it's okay (use ImportTestCase from lsst.utils to do that).

Reply:
The functionality is already clearly separated; that's the `show` function, and if there's nothing tricky about importing a `__main__` module (or an easy alternative) I'm happy to do that.

And I'm fine with making the `ArgumentParser` instance available for Sphinx purposes (per the other comment), but I think it's very important to keep bits of argument parser definition code close to the corresponding bits of minimal-main-implementation code that translates those CLI arguments to the form expected by the Python interfaces.  If you factor the other way, and keep all argument parser definition code together and keep all non-argument-parser-definition minimal-main-implementation code together instead, you cannot add structure to keep complexity under control.

Reply:
At the moment the `lsst.utils.tests.ImportTestCase` skips trying to import `__*.py` files so maybe I should explicitly only skip the `__init__.py` files. 

https://github.com/lsst/utils/blob/main/python/lsst/utils/tests.py#L388-L389

Reply:
I've updated the test code to run `__main__.main` (passing command-line arguments as a sequence of strings), so there's no need for import-only tests here anymore.